test/MusicStore.Test/AnnouncementComponentTest.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.Mvc.ModelBinding;
+using Microsoft.Data.Entity;
+using Microsoft.Framework.Caching.Memory;
+using Microsoft.Framework.DependencyInjection;
+using MusicStore.Models;
+using Xunit;
+
+namespace MusicStore.Components
+{
+    public class AnnouncementComponentTest
+    {
+        private readonly IServiceProvider _serviceProvider;
+
+        public AnnouncementComponentTest()
+        {
+            var services = new ServiceCollection();
+
+            services.AddEntityFramework()
+                      .AddInMemoryStore()
+                      .AddDbContext<MusicStoreContext>();
+
+            _serviceProvider = services.BuildServiceProvider();
+        }
+
+        [Fact]
+        public async Task AnnouncementComponent_Returns_LatestAlbum()
+        {
+            // Arrange
+            var viewContext = new ViewContext()
+            {
+                ViewData = new ViewDataDictionary(
+                    new EmptyModelMetadataProvider(),
+                    new ModelStateDictionary())
+            };
+
+            var announcementComponent = new AnnouncementComponent()
+            {
+                DbContext = _serviceProvider.GetRequiredService<MusicStoreContext>(),
+                Cache = _serviceProvider.GetRequiredService<IMemoryCache>(),
+                ViewContext = viewContext,
+                ViewData = new ViewDataDictionary(viewContext.ViewData),
+            };
+
+            var today = DateTime.UtcNow.Date;
```

Comment:
Hmm, this might be a case where the product code has to change to be unit testable. Code that uses DateTime.Now (or similar) cannot be unit tested. You need to use something like https://github.com/aspnet/Security/blob/dev/src/Microsoft.AspNet.Authentication/ISystemClock.cs to create a fake known time.

For example, I think this test could fail unexpectedly if the test run started at 11:59:59.9999pm and the date rolled over because it became 12:00:00.0000am the next day when the rest of the code executed, then the calculations would be off by 1 entire day.